### DISCUSSION TITLE
Desktop app could not connect on Linux under VPN

### DISCUSSION THREAD

**Discussion by mtgit19** [tutao/tutanota#4622] [Bugs]
I'm facing a connection issue after I have upgraded my OS from Debian 10 to Debian 11: the desktop app could not connect (and displaying a message: "Connection lost. Try again") when I'm connected to my work VPN. This happens even if I'm starting the app  with the proxy option: --proxy-server=http://proxy-host:proxy-port

    OS: Debian 11
    Desktop Environment: KDE Plasma 5.20.5
    Kernel Version: 5.10.0-18-amd64
    Tutanota Version: 3.100.2

the monkey has been patched
(node:27443) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `tutanota-desktop --trace-deprecation ...` to show where the warning was created)
version:   3.100.2
resolving key... tutanota-vault
...
Webapp ready
[27443:0930/140306.430573:ERROR:browser_main_loop.cc(271)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
autoUpdater info:
 Checking for update
...
[ProtocolProxy] error for GET https://mail.tutanota.com/rest/sys/session/Mj0lTIe--J-2/HHkaMgC77wQjVGeTBGLjcBn9Ozt29DS81mnwRrZnw2w:
[ProtocolProxy] dn [Error]: net operation failed with code -7
    at ClientRequest.<anonymous> (/tmp/.mount_tutanoOC3Ssh/resources/app.asar/desktop/DesktopMain.js:1:47208)
    at ClientRequest.emit (node:events:526:28)
    at TLSSocket.emitRequestTimeout (node:_http_client:758:9)
    at Object.onceWrapper (node:events:645:28)
    at TLSSocket.emit (node:events:538:35)
    at Socket._onTimeout (node:net:501:8)
    at listOnTimeout (node:internal/timers:559:17)
    at process.processTimers (node:internal/timers:502:7) {
  code: -7
}
[ProtocolProxy] {"name":"Error","message":"net operation failed with code -7","stack":"Error: net operation failed with code -7\n    at ClientRequest.<anonymous> (/tmp/.mount_tutanoOC3Ssh/resources/app.asar/desktop/DesktopMain.js:1:47208)\n    at ClientRequest.emit (node:events:526:28)\n    at TLSSocket.emitRequestTimeout (node:_http_client:758:9)\n    at Object.onceWrapper (node:events:645:28)\n    at TLSSocket.emit (node:events:538:35)\n    at Socket._onTimeout (node:net:501:8)\n    at listOnTimeout (node:internal/timers:559:17)\n    at process.processTimers (node:internal/timers:502:7)"}
[ProtocolProxy] failed after 20074ms

**Comment by charlag** [comment]
You are trying to pass proxy but I suspect that Electron is only applying it to the requests made from the web part. For some time we proxy all requests via native code which doesn't take proxy into account.

**Reply by mtgit19** [reply]
Thank you for your reply. 
Is there any way to make this work? Do I have other options to make the proxy available to the desktop app?

**Reply by charlag** [reply]
I don't think there's a way to make it work at the moment, sorry

**Reply by mtgit19** [reply]
Ok. Thank you for your help. 
Just wanted to add that I'm not tempted to use Tutanota desktop app. if I'm not able to connect behind a VPN. For security reasons...

**Reply by ghost** [reply]
Tbh, electron based services really don't need a dedicated application. It's literally worthless. Just enjoy it on your beloved browser.

**Reply by databasicsllc** [reply]
Liking the electron client, personally. On the proxy situation, there are some countries blocking tutanota at top level because it's so based and awesome. My employees have to use VPN in those places to get to our shiny new encrypted backend in Germany. I'd love to see proxy setting built into the app settings.
